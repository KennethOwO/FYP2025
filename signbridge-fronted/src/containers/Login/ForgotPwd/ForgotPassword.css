.sign-up-container {
  width: 500px;
  margin: 0 auto;
  text-align: center;
}

.sign-up-form {
  margin: 30px 0 50px 0;
  padding: 40px 20px 30px 20px;
  border: 1px solid #ccc;
  border-radius: 10px;
  background-color: #fff;
  box-shadow: 0 0 10px #000;
}

.sign-up-container h1 {
  font-size: 30px;
  margin: 20px 0 0 0;
  padding: 10px;
  border-bottom: #000 2px solid;
  width: 50%;
  margin-left: auto;
  margin-right: auto;
}
