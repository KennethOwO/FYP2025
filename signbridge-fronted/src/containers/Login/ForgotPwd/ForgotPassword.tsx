import React, { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import LoginInput from "../../../components/LoginInput/LoginInput";
import "./ForgotPassword.css";
import toast from "react-hot-toast";
import { useTranslation } from "react-i18next";
import Cookies from "js-cookie";

// @ts-ignore
import { auth } from "../../../../firebase";
import { sendPasswordResetEmail } from "firebase/auth";

const ForgotPassword: React.FC = () => {
	const { t } = useTranslation();
	const navigate = useNavigate(); // For the redirection
	const [loading, setLoading] = useState(false);

	const [email, setEmail] = useState("");
	const [emailError, setEmailError] = useState("");
	const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		setEmail(e.target.value);
		validateEmail(e.target.value);
	};

	// get token from cookies, if exist, redirect to home
	useEffect(() => {
		const token = Cookies?.get("token");
		if (token !== undefined) {
			navigate("/");
		}
	}, [navigate]);

	const validateEmail = (value: string) => {
		let error = "";
		const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
		if (!value.trim()) {
			error = t("email_required");
			setEmailError(t("email_required"));
			return error;
		} else if (!emailRegex.test(value)) {
			setEmailError(t("email_invalid"));
			return error;
		}
		setEmailError("");
		return error;
	};

	const handleSubmit = async (e: React.FormEvent<HTMLFormElement>): Promise<void> => {
		e.preventDefault();
		const emailErrors = validateEmail(email);

		let errorMessage = "";
		if (emailErrors.length > 0){
			errorMessage = emailErrors
		}

		if (emailErrors.length === 0) {
			setLoading(true);

			sendPasswordResetEmail(auth, email)
				.then(() => {
					toast.success(t("password_reset_link_sent"));
				})
				.catch(error => {
					toast.error(error.message);
				})
				.finally(() => {
					setLoading(false);
				});
		} else {
			toast.error(errorMessage);
		}
	};

	return (
		<div className="sign-up-container">
			<h1>{t("forgot_password_page")}</h1>
			<div className="sign-up-form">
				<form onSubmit={handleSubmit}>
					<LoginInput label={t("email")} type="email" placeholder="" value={email} onChange={handleEmailChange} error={emailError} />

					<button className="sign-up-btn" type="submit" disabled={loading}>
						{loading ? (
							<>
								<i className="fa fa-spinner fa-spin"></i> Loading
							</>
						) : (
							t("reset_password")
						)}
					</button>
				</form>
			</div>
		</div>
	);
};

export default ForgotPassword;
