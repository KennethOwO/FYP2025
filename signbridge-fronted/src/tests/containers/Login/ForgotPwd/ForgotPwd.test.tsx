import ForgotPassword from "@root/containers/Login/ForgotPwd/ForgotPassword";
import { create } from "react-test-renderer";
import { cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("Test ForgotPassword", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const tree = create(
      <MemoryRouter>
        <ForgotPassword />
      </MemoryRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
