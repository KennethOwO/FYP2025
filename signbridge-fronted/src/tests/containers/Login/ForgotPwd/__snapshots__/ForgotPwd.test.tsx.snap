// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ForgotPassword should render correctly 1`] = `
<div
  className="sign-up-container"
>
  <h1>
    Forgot Password
  </h1>
  <div
    className="sign-up-form"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="login-form-group "
      >
        <input
          onChange={[Function]}
          placeholder=""
          type="email"
          value=""
        />
        <label
          className="login-form-label"
          htmlFor="inp"
        >
          Email
        </label>
      </div>
      <button
        className="sign-up-btn"
        disabled={false}
        type="submit"
      >
        Reset Password
      </button>
    </form>
  </div>
</div>
`;
